src/Api/Vault/Controllers/CiphersController.cs

```diff
@@ -424,6 +424,59 @@ private async Task<bool> CanAccessUnassignedCiphersAsync(Guid organizationId)
         return false;
     }
 
+    /// <summary>
+    /// TODO: Move this to its own authorization handler or equivalent service - AC-2062
+    /// </summary>
+    private async Task<bool> CanModifyCipherCollectionsAsync(Guid organizationId, IEnumerable<Guid> cipherIds)
+    {
+        // If the user can edit all ciphers for the organization, just check they all belong to the org
+        if (await CanEditAllCiphersAsync(organizationId))
+        {
+            // TODO: This can likely be optimized to only query the requested ciphers and then checking they belong to the org
+            var orgCiphers = (await _cipherRepository.GetManyByOrganizationIdAsync(organizationId)).ToDictionary(c => c.Id);
+
+            // Ensure all requested ciphers are in orgCiphers
+            if (cipherIds.Any(c => !orgCiphers.ContainsKey(c)))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        // The user cannot access any ciphers for the organization, we're done
+        if (!await CanAccessOrganizationCiphersAsync(organizationId))
+        {
+            return false;
+        }
+
+        var userId = _userService.GetProperUserId(User).Value;
+        // Select all editable ciphers for this user belonging to the organization
+        var editableOrgCipherList = (await _cipherRepository.GetManyByUserIdAsync(userId, true))
```

Comment:
❓ `GetManyByUserIdAsync()` still sorts by `Edit` internally so we get a cipher with `ViewPassword == false` for one collection while another collection may have `ViewPassword == true`. If we happen to sort the `ViewPassword == false` first, then the user will get a authorization error.

I suspect we'll need to do something similar as the change we're making to the `CipherDetails_ReadByIdUserId` query.

Reply:
Fixed with [e498acb](https://github.com/bitwarden/server/pull/4844/commits/e498acb22ca03cf2d5be775b0b76e0c03e030483)